### ISSUE TITLE
Is it possible to get all places within an area (county or state) with ACS?

### ISSUE THREAD

**Issue by MaxGhenis** [cenpy-devs/cenpy#89]
Given ACS only has county and tract keys, it seems like not, but checking here.
```
acs = cenpy.products.ACS(2018)
acs._layer_lookup.keys()
```
>dict_keys(['county', 'tract'])

As expected, this produces an error (though I don't think it's the error message that's intended, given the [error code](https://github.com/cenpy-devs/cenpy/blob/master/cenpy/products.py#L504)):
```
acs.from_county('Ventura County, CA', level='place')
```

> ---------------------------------------------------------------------------
> IndexError                                Traceback (most recent call last)
> <ipython-input-115-27c9ea396e54> in <module>
> ----> 1 acs.from_county('Ventura County, CA', level='place')
> 
> ~/miniconda3/lib/python3.7/site-packages/cenpy/products.py in from_county(self, county, variables, level, **kwargs)
>     626     from_csa.__doc__ = _Product.from_place.__doc__.replace('place', 'CSA')
>     627     def from_county(self, county, variables=None, level='tract', **kwargs):
> --> 628         return self._from_name(county, variables, level, 'Counties', **kwargs)
>     629     from_county.__doc__ = _Product\
>     630                                     .from_place.__doc__\
> 
> ~/miniconda3/lib/python3.7/site-packages/cenpy/products.py in _from_name(self, place, variables, level, layername, return_geometry, cache_name, strict_within, return_bounds, geometry_precision)
>     586                                       'Try picking the state containing that level,'
>     587                                       ' and then selecting from that data after it is'
> --> 588                                       ' fetched'.format(level))
>     589         if level == 'block':
>     590             raise ValueError('The American Community Survey is only administered'
> 
> IndexError: tuple index out of range

**Comment by knaaptime** [comment]
you're mixing geographies here, asking for data from a county, but then including the `level=place` argument. In Census parlance, "place" basically means city. If you remove that argument, e.g.
![image](https://user-images.githubusercontent.com/4213368/71846820-52b8a080-3080-11ea-8d14-170cae4f30bf.png)

you get data for all tracts in ventura county. Is that what you're after?

**Comment by knaaptime** [comment]
(`place`s don't nest with other FIPS geographies and ACS isn't tabulated to the place geography, which is why you're getting the error that ACS isnt supporter for that query)

**Comment by knaaptime** [comment]
reading closer, i guess you're trying to grab all the census designated places in ventura? that's not supported yet, but the universe of CDPs isn't that big. You might just grab all the [data for CA](https://www2.census.gov/geo/tiger/TIGER2019/PLACE/tl_2019_06_place.zip) and filter it down

**Comment by MaxGhenis** [comment]
Is it possible to get data at the place level, or is cenpy only by tract/block today? Even just a single row when calling `from_place` would be fine.

**Comment by knaaptime** [comment]
the simplest way, currently, would be to do a place query like

`cenpy.products.ACS(2018).from_place(PLACE)`

which will give you back the tract-level data for the place, then you could do a [dissolve](http://geopandas.org/aggregation_with_dissolve.html) which would effectively give you place-level data. You'll have to recalculate percents and what not, but i think that would get you what you're after?

**Comment by MaxGhenis** [comment]
Don't tract sums often not add up to higher levels, due to Census-side data cleaning and privacy mechanisms?

**Comment by knaaptime** [comment]
im pretty sure that's not the case. the way the FIPS system encodes nesting, you should be able to aggregate up the hierarchy without issue

now, as i said before, `places` are a somewhat different beast because their boundaries dont always align with other census geographies. If you have tracts that cross the place boundaries, you'll obviously have a bit of error (and if you're using ACS data, those margins can be pretty wide). So in the case of places, its possible that your tract-level aggregation wouldn't match the stats tabulated by the official place boundary, but you'd still get a pretty reasonable estimate

**Comment by MaxGhenis** [comment]
Aggregating is possible, but you won't always get the same result as the census.

For example, I exported income data from data.census.gov for each Ventura County tract, as well as Ventura County overall. Here's the [query](https://data.census.gov/cedsci/table?q=income&g=0500000US06111.140000,06111&hidePreview=true&table=S1901&tid=ACSST5Y2018.S1901&t=Income%20%28Households,%20Families,%20Individuals%29&vintage=2018&cid=S1901_C01_001E&layer=county&lastDisplayedRow=16&moe=false&tp=true) and [spreadsheet](https://docs.google.com/spreadsheets/d/1tVavBqxcxezXIo0SsR-uuBrV4sC5NQobhptEzKE66m8/edit#gid=1285982309).

The counts all match, e.g. summing households and families by tract equals the total. But when recalculating percentages, you get issues. Summing up the number of families with <$10k income (by row, multiplying families by the percentage) produces an error of 0.1%, which looks like it could result from two tracts that don't have income distribution data. I imagine the difference could be bigger for other geos and/or fields. Medians also can't be reconstructed.

Is this an API issue or just hasn't been implemented here yet? Looks like tidycensus can pull out other levels.

**Comment by knaaptime** [comment]
should be possible, but i'm getting `error: unknown/unsupported geography heirarchy` if i try to use a place query like `https://api.census.gov/data/2017/acs/acs5?get=NAME,group(B01001)&for=county:073&in=state:01+place:07000` (as shown in the [api docs](https://www.census.gov/content/dam/Census/data/developers/api-user-guide/api-guide.pdf)). 

Everything works fine if i remove the place query. Any idea if i'm missing something obvious @walkerke?

**Comment by ljwolf** [comment]
Yeah, this would involve adding `place` to the `_layer_lookup` and checking what, if anything, needs to change in how the results from that are parsed. So, no, not currently supported in cenpy, but definitely should be do-able!

**Comment by walkerke** [comment]
@knaaptime Census changed queries that cross boundaries like that by including the `(or part)` designation, like this: 

https://api.census.gov/data/2017/acs/acs5?get=NAME,group(B01001)&for=county%20(or%20part):073&in=state:01+place:07000

That query gets you data for the part of Birmingham that is located within Jefferson County (but not in Shelby County).  

That specific configuration isn't supported in tidycensus but I believe is do-able with censusapi.

**Comment by knaaptime** [comment]
ahh, sweet. thanks Kyle!
